solidity/contracts/igps/InterchainGasPaymaster.sol

```diff
@@ -68,9 +74,34 @@ contract InterchainGasPaymaster is IInterchainGasPaymaster, OwnableUpgradeable {
             require(_success, "Interchain gas payment refund failed");
         }
 
-        emit GasPayment(_messageId, _gasAmount, msg.value);
+        emit GasPayment(_messageId, _gasAmount, _requiredPayment);
+    }
+
+    /**
+     * @notice Sets the gas oracle for a remote domain.
+     * @param _remoteDomain The remote domain.
+     * @param _gasOracle The gas oracle.
+     */
+    function setGasOracle(uint32 _remoteDomain, address _gasOracle)
+        external
+        onlyOwner
+    {
+        gasOracles[_remoteDomain] = IGasOracle(_gasOracle);
+        emit GasOracleSet(_remoteDomain, _gasOracle);
+    }
+
+    /**
+     * @notice Transfers the entire native token balance to the owner of the contract.
+     * @dev The owner must be able to receive native tokens.
+     */
+    function claim() external {
```

Comment:
why a diff here?

Reply:
Just moved it to be with the other external functions, no actual code changes